File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/parameters/ProductClient.java
package io.micronaut.rabbitmq.docs.parameters;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.rabbitmq.annotation.Binding;
import io.micronaut.rabbitmq.annotation.RabbitClient;
// end::imports[]

@Requires(property = "spec.name", value = "BindingSpec")
// tag::clazz[]
@RabbitClient
public interface ProductClient {

    @Binding("product") // <1>
    void send(byte[] data);

    void send(@Binding String binding, byte[] data); // <2>
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/rabbitmq/docs/parameters/ProductClient.kt
package io.micronaut.rabbitmq.docs.parameters

// tag::imports[]
import io.micronaut.rabbitmq.annotation.Binding
import io.micronaut.rabbitmq.annotation.RabbitClient
import io.micronaut.context.annotation.Requires
// end::imports[]

@Requires(property = "spec.name", value = "BindingSpec")
// tag::clazz[]
@RabbitClient
interface ProductClient {

    @Binding("product") // <1>
    fun send(data: ByteArray)

    fun send(@Binding binding: String, data: ByteArray)  // <2>
}
// end::clazz[]